### PR TITLE
Align ORPO with DPO: support iterable and dict eval datasets

### PR DESCRIPTION
**Pull request by DaoyuanLi2816**
## Status

Closed to consolidate related trainer work instead of keeping several overlapping pull requests open.

## AI writing disclosure

- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.

### CODE REVIEW

File: trl/experimental/orpo/orpo_trainer.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 from accelerate import PartialState
 from accelerate.logging import get_logger
 from accelerate.utils import is_peft_model
-from datasets import Dataset
+from datasets import Dataset, IterableDataset
```
[the remark this answers is not in the document]
Comment (DaoyuanLi2816) [reply trl/experimental/orpo/orpo_trainer.py]: Good catch — added the same guard the other trainers (DPO/KTO/TPO/Reward/SFT) already have: when `train_dataset` is an `IterableDataset`, `dispatch_batches` is forced to `False` (with a warning if it was explicitly set to `True`).

**Comment by DaoyuanLi2816** [comment]
Friendly bump on this one — it's been mergeable and CI-green since it was opened, with the earlier Bugbot finding (the missing `dispatch_batches` guard for `IterableDataset`) addressed. It extends the same iterable / dict eval-dataset handling that recently landed on the core trainers (#6322, #6326) to the experimental `ORPOTrainer`, following the same alignment pattern as the previously merged #6087 and #6192. Happy to rebase or adjust if anything's needed.

**Comment by DaoyuanLi2816** [comment]
Closing this to consolidate the outstanding trainer-parity work instead of keeping several related PRs open in parallel.
